### PR TITLE
ci: Upgrade taskgraph to 7.3.0

### PR DESCRIPTION
**Pull request by hneiva**
## Description

    Upgrade taskgraph to 7.3.0

    - Use the new taskgraph tagged image
    - Recompiled requirements to pick up new mozilla-taskgraph dependencies

## Checklist
    
- [X] My code follows the style guidelines for this project
- [X] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [X] I have performed a self review of my own code
- [X] I have commented my code PARTICULARLY in hard to understand areas
- [X] I have added thorough tests where needed

### CODE REVIEW

**Comment by hneiva** [comment]
Ran `taskgraph full --diff -p taskcluster/test/params/*` and all tasks were identical.

File: taskcluster/requirements.in [hunk 1]
```diff
@@ -1,5 +1,4 @@
 # For instructions on managing dependencies, see:
 # https://taskcluster-taskgraph.readthedocs.io/en/latest/howto/bootstrap-taskgraph.html
 
-taskcluster-taskgraph >= 7.0.2
```
Comment (bhearsum) [inline taskcluster/requirements.in:4]: There's a bunch of explicit references to `taskgraph`; you should keep this specified as an explicit dependency.

**Review by bhearsum — CHANGES_REQUESTED** [review]
r=me with the `taskcluster-taskgraph` dependency re-added to `requirements.in`.

File: taskcluster/requirements.in [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
-mozilla-taskgraph >= 1.1.0
+mozilla-taskgraph==1.2.2
+taskcluster-taskgraph==7.3.0
```
Comment (bhearsum) [inline taskcluster/requirements.in:5]: Do you want `==` or `>=`?

**Review by bhearsum — APPROVED** [review]

File: taskcluster/requirements.in (hunk 2 above)
Comment (hneiva) [reply taskcluster/requirements.in:5]: I think we should keep `==` so dependabot won't try to update it..? Thoughts?

**Pull request merged**